### ISSUE TITLE
Forward LabMessages to responsible health department via SORMAS2SORMAS [3]

### ISSUE THREAD

**Issue by FredrikSchaeferVitagroup** [hzi-braunschweig/SORMAS-Project#3742]
### Situation Description

SORMAS will be able do import lab messages from DEMIS soon.
Theoretically, when a health department requests new lab messages from demis, it should only get messages that this specific health department is responsible for. In practice, a health department also get's lab messages for persons it is not responsible for. The lab message then needs to be forwarded to the responsible health department via SORMAS2SORMAS then.

The responsibility for a person is regulated in [§11 of the german Infektionsschutzgesetz](http://www.gesetze-im-internet.de/ifsg/__11.html). There, it is stated:
`(3) Für die Vervollständigung, Zusammenführung und Übermittlung der Daten nach Absatz 1 ist das Gesundheitsamt zuständig, in dessen Bezirk die betroffene Person ihre Hauptwohnung hat oder zuletzt hatte. Falls ein Hauptwohnsitz nicht feststellbar ist oder die betroffene Person sich dort gewöhnlich nicht aufhält, so ist das Gesundheitsamt zuständig, in dessen Bezirk sich die betroffene Person gewöhnlich aufhält. Falls ein solcher Aufenthaltsort nicht feststellbar ist oder in den Fällen der Meldung nach § 6 Absatz 3 Satz 1 ist das Gesundheitsamt zuständig, welches die Daten erstmals verarbeitet hat. Das nach den Sätzen 1 bis 3 zuständige Gesundheitsamt kann diese Zuständigkeit an ein anderes Gesundheitsamt mit dessen Zustimmung abgeben, insbesondere wenn schwerpunktmäßig im Zuständigkeitsbereich des anderen Gesundheitsamtes weitere Ermittlungen nach § 25 Absatz 1 angestellt werden müssen.`

[This tool from RKI](https://tools.rki.de/PLZTool/?) indicates that if there is a (main) postal code existing for a person of a lab message, it should be possible to determine the responsible health department. 

### Feature Description

When processing a lab message, SORMAS should first check whether the `labMessage.personPostalCode` is not null. If that is the case, SORMAS should check whether this particular SORMAS instance belongs to the responsible health department. If not, the according SORMAS instance should automatically be determined via the `labMessage.personPostalCode`. The processing user should be informed that his health department probably is not responsible for this labMessage. If the determination of the responsible SORMAS instance was succesfull, forwarding to this instance via SORMAS2SORMAS should be proposed.

If the `labMessage.personPostalCode` is null, the determination of the responsible SORMAS instance should be tried again once the labMessage is related to a person.

Open questions:

- How do we relate a postal code of the lab message person to a sormas instance? HZI proposed to do the determination and relation of resposible health departments via the municipality keys (Gemeindeschlüssel). Is this a feasible approach? Does it have any other usages, e.g. for the SurvNet interface?

- What is to be done with a forwarded lab message in the forwarding SORMAS instance? Does it need to be deleted? Should we have a third status (processed, unprocessed, forwarded)?

### Additional Information

**in scope:**
send lab-message to other SORMAS instance 
When successfully sent, we delete it on the sender's site.

Prerequisite: The receiving SORMAS Instance has the same Version of the DEMIS adapter

**out of scope:**
[ ] Add responsible postal code to the SORMAS Instance if they are using SORMAS X
[ ] Check wether the postal code of the lab message can be mapped to the receiving SORMAS instance or another SORMAS X Instance that has SORMAS 2 SORMAS enabled
[ ] Propose the responsible Health Department by the city codes
[ ] DEMIS Adapter should be installed by default
[ ] Manage the postal codes as an admin in SORMAS-UI Configuration 

**Proposal**
Get a list of all German Health Departments with their responsible Postal Codes and add this to the infrastructure data like region, district.

**Comment by FredrikSchaeferVitagroup** [comment]
Some part of the out of scope topics may be covered in https://github.com/hzi-braunschweig/SORMAS-Project/issues/4508

**Comment by FredrikSchaeferVitagroup** [comment]
Some advancements have been implemented on DEMIS site: https://confluence-demis.gematik.de/display/DSKB/Meldungsrouting
(seems to be only available in german).
